lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -221,17 +223,32 @@ defmodule Mix.Compilers.Elixir do
         dest
       )
 
-    {modules, exports, changed, sources_stats}
+    {modules, exports, changed, sources_snapshots}
   end
 
-  defp mtimes_and_sizes(sources) do
+  defp snapshots(sources) do
     Enum.reduce(sources, %{}, fn source(source: source, external: external), map ->
       Enum.reduce([source | external], map, fn file, map ->
-        Map.put_new_lazy(map, file, fn -> Mix.Utils.last_modified_and_size(file) end)
+        Map.put_new_lazy(map, file, fn -> snapshot(file) end)
       end)
     end)
   end
 
+  defp digest(file) do
+    case File.read(file) do
+      {:ok, content} ->
+        :crypto.hash(:md5, content)
```

Comment:
```suggestion
        :erlang.md5(content)
```

The difference is that `:crypto` is a separate application, that we would need to start and depend on, but luckily MD5 is part of Erlang too.